ui-ngx/src/app/shared/components/rule-chain/rule-chain-select-panel.component.ts

```diff
@@ -0,0 +1,143 @@
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  Inject,
+  InjectionToken,
+  OnDestroy,
+  ViewChild
+} from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Observable, of, Subject } from 'rxjs';
+import { catchError, debounceTime, distinctUntilChanged, map, switchMap, takeUntil, startWith, share } from 'rxjs/operators';
+import { PageLink } from '@shared/models/page/page-link';
+import { Direction } from '@shared/models/page/sort-order';
+import { emptyPageData, PageData } from '@shared/models/page/page-data';
+import { OverlayRef } from '@angular/cdk/overlay';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { RuleChain, RuleChainType } from '@shared/models/rule-chain.models';
+import { RuleChainService } from '@core/http/rule-chain.service';
+import { TranslateService } from '@ngx-translate/core';
+
+export const RULE_CHAIN_SELECT_PANEL_DATA = new InjectionToken<any>('RuleChainSelectPanelData');
+
+export interface RuleChainSelectPanelData {
+  ruleChainId: string | null;
+  ruleChainType: RuleChainType;
+}
+
+@Component({
+  selector: 'tb-rule-chain-select-panel',
+  templateUrl: './rule-chain-select-panel.component.html',
+  styleUrls: ['./rule-chain-select-panel.component.scss']
+})
+export class RuleChainSelectPanelComponent implements AfterViewInit, OnDestroy {
+
+  ruleChainId: string;
+  ruleChainType: RuleChainType;
+
+  selectRuleChainGroup: FormGroup;
+
+  @ViewChild('ruleChainInput', {static: true}) userInput: ElementRef;
+
+  filteredRuleChains: Observable<Array<RuleChain>>;
+
+  searchText = '';
+
+  ruleChainSelected = false;
+
+  result?: RuleChain;
+
+  private dirty = false;
+  private destroy$ = new Subject<void>();
+
+  constructor(@Inject(RULE_CHAIN_SELECT_PANEL_DATA) public data: RuleChainSelectPanelData,
+              public overlayRef: OverlayRef,
+              public translate: TranslateService,
+              private fb: FormBuilder,
+              private ruleChainService: RuleChainService) {
+    this.ruleChainId = data.ruleChainId;
+    this.ruleChainType = data.ruleChainType;
+    this.selectRuleChainGroup = this.fb.group({
+      ruleChainInput: ['', {nonNullable: true}]
+    });
+    this.filteredRuleChains = this.selectRuleChainGroup.get('ruleChainInput').valueChanges
+      .pipe(
+        debounceTime(150),
+        startWith(''),
+        distinctUntilChanged((a: string, b: string) => a.trim() === b.trim()),
+        switchMap(name => this.fetchRuleChains(name)),
+        share(),
+        takeUntil(this.destroy$)
```

Comment:
This operator does not make sense because the control is done through the template and not manually.